### PR TITLE
Nan 226 payload ordering

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Order a payload by creation

## Issue ticket number and link
NAN-226 & NAN-170

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts [hunk 1]
```diff
@@ -24,11 +24,27 @@ export default async function route(nango: Nango, integration: ProviderConfig, h
     }
 
     if (Array.isArray(body)) {
-        const sorted = body.sort((a, b) => {
-            return a.occurredAt - b.occurredAt;
-        });
-        for (const event of sorted) {
-            await nango.executeScriptForWebhooks(integration, event, 'subscriptionType', 'portalId');
+        const groupedByObjectId = body.reduce((acc, event) => {
+            (acc[event.objectId] = acc[event.objectId] || []).push(event);
```
Comment (TBonnin) [inline packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts:28]: I don't understand this line, especially the `acc[event.objectId] = acc[event.objectId]`

File: packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts [hunk 2]
```diff
@@ -24,11 +24,27 @@ export default async function route(nango: Nango, integration: ProviderConfig, h
...
+            const sorted = groupedByObjectId[objectId].sort((a: any, b: any) => {
+                const aIsCreation = a.subscriptionType.endsWith('.creation');
+                const bIsCreation = b.subscriptionType.endsWith('.creation');
+                if (aIsCreation && !bIsCreation) {
+                    return -1;
+                }
+                if (!aIsCreation && bIsCreation) {
+                    return 1;
+                }
+                return a.occurredAt - b.occurredAt;
```
Comment (TBonnin) [inline packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts]: would something like the following simpler?
```
const aIsCreation = a.subscriptionType.endsWith('.creation') ? 1 : 0;
const bIsCreation = b.subscriptionType.endsWith('.creation') ? 1 : 0;
return bIsCreation - aIsCreation || a.occurredAt - b.occurredAt;
```

also maybe having a test for this function might be useful

Comment (khaliqgant) [reply packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts]: Thanks, updated with https://github.com/NangoHQ/nango/pull/1529/commits/ad0995ffc09df6676478dcc69a4927a6a0344e6a

File: packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts:28]: Orders the payload by eventId

**Pull request merged**